File: RedSheet.DbEntities/Models/LiteMeetingPlanning.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("LiteMeetingPlannings", Schema = "dbo")]
	[Serializable]
	public partial class LiteMeetingPlanning  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int LiteMeetingPlanningId { get; set; } 
 
	[Required] 
	[MaxLength(350)]
	public string LiteMeetingPlanningValue { get; set; } 
 
	[RelationshipTableAttribue("LiteMeetingManagements","dbo")]
	//Relationships
	public int LiteMeetingManagementId { get; set; } 
 
	[ForeignKey("LiteMeetingManagementId")]
	public LiteMeetingManagement LiteMeetingManagement { get; set; } 

	public LiteMeetingPlanning()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/lite-meeting-planning.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { LiteMeetingManagement,  vLiteMeetingPlanningRecord  } from './'
export class LiteMeetingPlanning {
    constructor(liteMeetingPlanning?: LiteMeetingPlanning  | vLiteMeetingPlanningRecord )  {
        let properties = [ "liteMeetingPlanningId", "liteMeetingPlanningValue", "liteMeetingManagementId",];
        for (let property of properties)
            if (liteMeetingPlanning && liteMeetingPlanning[property])
                this[property] = liteMeetingPlanning[property];
    }
 
	liteMeetingPlanningId : number =   0 ;
 
    @required()
    @maxLength(350)
	liteMeetingPlanningValue : string =   undefined;
 
    @range(0,2147483647)
	liteMeetingManagementId : number =   undefined;
	liteMeetingManagement : LiteMeetingManagement  ;


}
